indico/modules/events/registration/controllers/display.py

```diff
@@ -0,0 +1,93 @@
+from indico.modules.events.registration.models.registrations import Registration
+from indico.modules.events.registration.util import get_event_section_data, make_registration_form
+from indico.modules.events.registration.views import (WPDisplayRegistrationFormConference,
+                                                      WPDisplayRegistrationFormMeeting,
+                                                      WPDisplayRegistrationFormLecture)
+from indico.modules.payment import event_settings
+from indico.web.flask.util import url_for
+from MaKaC.webinterface.rh.conferenceDisplay import RHConferenceBaseDisplay
+
+
+class RHRegistrationFormDisplayBase(RHConferenceBaseDisplay):
+    def _checkParams(self, params):
+        RHConferenceBaseDisplay._checkParams(self, params)
+        self.event = self._conf
+
+    @property
+    def view_class(self):
+        mapping = {
+            'conference': WPDisplayRegistrationFormConference,
+            'meeting': WPDisplayRegistrationFormMeeting,
+            'simple_event': WPDisplayRegistrationFormLecture
+        }
+        return mapping[self.event.getType()]
+
+
+class RHRegistrationFormList(RHRegistrationFormDisplayBase):
+    """List of all registration forms in the event"""
+
+    def _process(self):
+        regforms = RegistrationForm.find_all(is_active=True, event_id=int(self.event.id))
+        return self.view_class.render_template('display/regforms_list.html', self.event, regforms=regforms,
+                                               event=self.event)
+
+
+class RHRegistrationFormSubmit(RHRegistrationFormDisplayBase):
+    """Submit a registration form"""
+
+    CSRF_ENABLED = True
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.regform
+        }
+    }
+
+    def _checkParams(self, params):
+        RHRegistrationFormDisplayBase._checkParams(self, params)
+        self.regform = RegistrationForm.find_one(id=request.view_args['reg_form_id'])
+
+    def _process_POST(self):
+        form = make_registration_form(self.regform)()
+        if form.validate_on_submit():
+            self._save_registration(form.data)
+            return redirect(url_for('.display_regforms_list', self.regform))
+        return self.view_class.render_template('display/regform_display.html', self.event, event=self.event,
+                                               sections=get_event_section_data(self.regform), regform=self.regform,
+                                               currency=event_settings.get(self.event, 'currency'))
+
+    def _save_registration(self, data):
+        registration = Registration(user=session.user, registration_form=self.regform)
+        db.session.add(registration)
+        for form_item in self.regform.form_items:
+            value = data.get('*genfield*{0}-{1}'.format(form_item.parent_id, form_item.id), None)
+            if value:
+                form_item.wtf_field.save_data(registration, value)
```

Comment:
Wouldn't it be better to always call this even if there's no value? A checkbox should probably be stored as `false` not `''` or `null` if it's not checked.